blog/_posts/2025-09-05-pair-llmd-and-rag-on-aks.md

```diff
@@ -0,0 +1,177 @@
+|--|--|--|
+|Prefill/Decode (P/D) Disaggregation|Separates the compute-heavy prefill stage (KV cache building) from the decode stage (autoregressive token generation) on dedicated GPUs, where each GPU pool can be independently scheduled and scaled|Optimizes throughput for long contexts; prevents resource contention for concurrent requests when processing long RAG queries; enables flexible scaling and lowers time-to-first-token (TTFT) and improves overall token output time (TPOT)|
+|Intelligent routing (via Gateway API Inference Extension's Endpoint Picker)|Schedules requests based not only on server queue length and available KV cache size, but also prefix cache hit probability. Maximize the reuse of KV cache for queries with overlapping system prompts and common context retrieved from the vector store|Minimizes latency; enables efficient reuse of KV cache for similar or repeated contexts; handles RAG queries of mixed context length and structure robustly|
+|Disaggregated KV cache management|Offloads KV cache across local or remote stores, with orchestration from decode-sidecar and advanced cache eviction policies for efficient memory use|Supports much longer input contexts and multiple concurrent sessions with reduced GPU memory overhead, enabling scalability for large RAG pipelines|
+
+## Let’s get started: KAITO RAGEngine backed by llm-d with P/D Disaggregation
+
+With the [prerequisites](https://github.com/kaito-project/kaito-cookbook/tree/master/examples/ragengine-llm-d#prerequisites) covered, this guide will dive into the creation of two distinct but related endpoints:
+
+1. **Inference Endpoint**: an OpenAI API compatible inference service in Kubernetes, created by the llm-d stack. Jump to the [llm-d inference endpoint](https://github.com/kaito-project/kaito-cookbook/tree/master/examples/ragengine-llm-d#inference-endpoint) section in our GitHub repository to set it up.
+
+1. **RAG Endpoint**: a RAG service provisioned by KAITO with the inference endpoint pointed to step (1) for users to efficiently index and query their documents. Check out the steps in this [RAG endpoint](https://github.com/kaito-project/kaito-cookbook/tree/master/examples/ragengine-llm-d#rag-endpoint) GitHub repo section to spin up and verify your RAGEngine workspace, which includes a YAML manifest that looks like:
+
+```bash
+apiVersion: kaito.sh/v1alpha1 
+kind: RAGEngine 
+metadata: 
+  name: ragengine-llm-d 
+spec: 
+  compute: 
+    instanceType: "Standard_DS2_v2" 
+    labelSelector: 
+      matchLabels: 
+        node.kubernetes.io/instance-type: Standard_DS2_v2
+  embedding:
+    local:
+      modelID: "BAAI/bge-small-en-v1.5"
+  inferenceService:
+    url: "http://<inference-url>/v1/chat/completions"
+```
+
+> **Note**: This RAG service only requires general-purpose compute, like the Azure `Standard_DS2_v2` SKU shown above, which often provides a more cost-effective and accessible alternative to the GPU-intensive process of continuous fine-tuning.
```

Comment:
I believe Standard_DS2_v4 are more readily available these days over v2 so maybe we should use that instead.